### PR TITLE
add basic typings for the 'isaac' package

### PR DESCRIPTION
**Pull request by mo**
- [x] Test the change
- [x] Add or edit tests to reflect the change
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Adding typings for npm package "isaac":

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Comment by mo** [comment]
@plantain-00 Thanks for the review. This is the first time I write typings for third party library and I'm unsure how to resolve the issue. Could you give some advice?

I have tried to switch the last line of the definition file to "export = isaac;" instead, but if I do this then I get an error when I try to use the typing. In particular, when I do:

```import isaac from 'isaac'```

...then I get the error:

```(4,8): Module '"/home/molsson/dv/isaac-typescript-test-project/node_modules/@types/isaac/index"' has no default export.```

**Comment by mo** [comment]
When I'm coding in vanilla Javascript I can do:

```
import isaac from 'isaac';
console.log(isaac.random());
```

...and in JS that just works.

**Comment by mo** [comment]
Thanks for the clarification. I updated the typings to use "export =" and "import isaac = require('isaac')" now.

I was also able to use them in my test project once I added the "allowSyntheticDefaultImports": true to my tsconfig. I used this instead of --esModuleInterop because I read a tweet that @DanielRosenwasser wrote here: https://twitter.com/drosenwasser/status/1003097042653073408 

I hope I got everything right this time!

File: types/isaac/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+declare namespace isaac {
+    function internals(): any;
+    function prng(n: number): void;
+    function rand(): number;
+    function random(): number;
+    function reset(): void;
+    function seed(s: string | number | ReadonlyArray<number>): void;
+}
+
+export default isaac;
```
[the remark this answers is not in the document]
Comment (mo) [reply types/isaac/index.d.ts]: hopefully this is fixed now

**Review by plantain-00 — APPROVED** [review]

File: types/isaac/isaac-tests.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import isaac = require('isaac');
```
Comment (sandersn) [inline types/isaac/isaac-tests.ts]: the exports are ES syntax, so the import should use ES syntax: `import * as isaac from 'isaac'`

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/isaac/isaac-tests.ts (hunk 2 above)
Comment (mo) [reply types/isaac/isaac-tests.ts]: fixed now

**Comment by sandersn** [comment]
The CI failure is not related and is now fixed in master. I'll go ahead and merge.

**Pull request merged**